docs/book/src/clusterctl/provider-contract.md

```diff
@@ -150,8 +150,32 @@ will look for objects to reconcile.
 
 #### Variables
 
-The components YAML can contain environment variables matching the regexp `\${\s*([A-Z0-9_]+)\s*}`; it is highly
-recommended to prefix the variable name with the provider name e.g. `${ AWS_CREDENTIALS }`
+The components YAML can contain environment variables matching the format ${VAR}; it is highly
+recommended to prefix the variable name with the provider name e.g. `${AWS_CREDENTIALS}`
+
+<aside class="note warning">
+
+<h1>Warning</h1>
+
+`clusterctl` supports variables with leading/trailing spaces and other meta
+characters such as: `${ VAR }`, `${ VAR}`,`${VAR }` and `${VAR$FOO}`.
```

Comment:
it might be confusing to readers that it says here that clusterctl supports `${VAR$FOO}` and then below say it is no longer supported

Reply:
Thanks for the feedback! I'll try and clarify this a bit more.